File: index.server/Index.server/src/main/java/br/com/pedroxsqueiroz/Index/server/dtos/ConceptDto.java
package br.com.pedroxsqueiroz.Index.server.dtos;


import br.com.pedroxsqueiroz.Index.concept.models.Concept;
import br.com.pedroxsqueiroz.Index.concept.models.ContentChunck;
import lombok.Getter;

import java.util.List;
import java.util.Set;

@Getter
public class ConceptDto {

    public ConceptDto(Concept concept) {
        this.id = concept.getId();
        this.name = concept.getName();
        this.description = concept.getDescription();

        Set<ContentChunck> chuncks = concept.getChuncks();
        if(chuncks != null)
        {
            this.chuncks = chuncks.stream().map(ContentChunckDto::new).toList();
        }
    }

    public ConceptDto(Concept concept, List<ContentChunckDto> chuncks) {
        this(concept);
        this.chuncks = chuncks;
    }

    private String id;
    private String name;
    private String description;
    private List<ContentChunckDto> chuncks;
}

File: index.web_cli/src/dtos/concept-dto.ts
import ContentChunckDto from "./content-chunck-dto";

export default class ConceptDto{

    constructor(
        id:string,
        name:string,
        descrption:string,
        chuncks: ContentChunckDto[]
    ){
        this.id = id;
        this.name = name;
        this.descrption = descrption;
        this.chuncks = chuncks;
    }

    id:string ;
    name:string ;
    descrption:string ;
    chuncks: ContentChunckDto[] ;

}